consensus/XDPoS/engines/engine_v2/timeout.go

```diff
@@ -92,34 +96,39 @@ func (x *XDPoS_v2) verifyTC(chain consensus.ChainReader, timeoutCert *types.Time
 		return fmt.Errorf("Empty master node lists from snapshot")
 	}
 
-	if timeoutCert == nil {
-		log.Warn("[verifyTC] TC is Nil")
-		return utils.ErrInvalidTC
-	} else if timeoutCert.Signatures == nil || (len(timeoutCert.Signatures) < x.config.V2.CurrentConfig.CertThreshold) {
-		log.Warn("[verifyTC] Invalid TC Signature is nil or empty", "timeoutCert.Round", timeoutCert.Round, "timeoutCert.GapNumber", timeoutCert.GapNumber, "Signatures len", len(timeoutCert.Signatures))
-		return utils.ErrInvalidTC
+	signatures, duplicates := UniqueSignatures(timeoutCert.Signatures)
+	if len(duplicates) != 0 {
+		for _, d := range duplicates {
+			log.Warn("[verifyQC] duplicated signature in QC", "duplicate", common.Bytes2Hex(d))
+		}
+	}
```

Comment:
duplication won't really hurt the whole functionality, preventing multiple nodes with same key with same signatures. And it will check the number of signatures later.